src/app/models/submission.server.model.ts

```diff
@@ -285,14 +256,34 @@ EncryptSubmissionSchema.statics.findSingleMetadata = function (
   formId: string,
   submissionId: string,
 ): Promise<StorageModeSubmissionMetadata | null> {
-  const pageResults: Promise<MetadataAggregateResult[]> = this.aggregate(
-    createAggregateQueryWithPaymentsProjectionParams({
-      form: mongoose.Types.ObjectId(formId),
-      _id: mongoose.Types.ObjectId(submissionId),
-    }),
-  )
-    .limit(1)
-    .exec()
+  const pageResults: Promise<MetadataAggregateResult[]> = this.aggregate([
+    {
+      $match: {
+        submissionType: SubmissionType.Encrypt,
+        form: mongoose.Types.ObjectId(formId),
+        _id: mongoose.Types.ObjectId(submissionId),
+      },
+    },
+    { $limit: 1 },
```

Comment:
why do we need `{ $limit: 1 }` here; wouldn't there be at most 1 submission document returned for  a given submission id?

Reply:
The query would continue to search for more even if we know that there should only be one -- it's still O(logn) through the index so theoretically we can remove this, but I feel that it's ok to leave it in incase we added some bad queries that prevented the use of index